manifests/runner.pp

```diff
@@ -1,68 +1,145 @@
+# == Define: gitlab_ci_multi_runner::runner
+#
+# Define for creating a gitlab-ci runner.
+#
+# gitlab_ci_multi_runner can/should be included to install
+# gitlab-ci-multi-runner if needed.
+#
+# === Parameters
+#
+# [*gitlab_ci_url*]
+#   URL of the Gitlab Server.
+#   Default: undef.
+#
+# [*tags*]
+#   Array of tags.
+#   Default: undef.
+#
+# [*token*]
+#   CI Token.
+#   Default: undef.
+#
+# [*executor*]
+#   Executor - Shell, parallels, ssh, docker etc.
+#   Default: undef.
+#
+# [*docker_image*]
+#   The Docker Image (eg. ruby:2.1).
+#   Default: undef.
+#
+# [*docker_privileged*]
+#   Run Docker containers in privileged mode.
+#   Default: undef.
+#
+# [*docker_mysql*]
+#   MySQL version (X.Y) or latest.
+#   Default: undef.
+#
+# [*docker_postgres*]
+#   Postgres version (X.Y) or latest.
+#   Default: undef.
+#
+# [*docker_redis*]
+#   Redis version (X.Y) or latest.
+#   Default: undef.
+#
+# [*docker_mongo*]
+#   Mongo version (X.Y) or latest.
+#   Default: undef.
+#
+# [*parallels_vm*]
+#   The Parallels VM (eg. my-vm).
+#   Default: undef.
+#
+# [*ssh_host*]
+#   The SSH Server Address.
+#   Default: undef.
+#
+# [*ssh_port*]
+#   The SSH Server Port.
+#   Default: undef.
+#
+# [*ssh_user*]
+#   The SSH User.
+#   Default: undef.
+#
+# [*ssh_password*]
+#   The SSH Password.
+#   Default: undef.
+#
+# [*require*]
+#   Array of requirements for the runner registration resource.
+#   Default: [ Class['gitlab_ci_multi_runner'] ].
+#
+# === Examples
+#
+#  gitlab_ci_multi_runner::runner { "This is My Runner":
+#      gitlab_ci_url => 'http://ci.gitlab.examplecorp.com'
+#      tags          => ['tag', 'tag2','java', 'php'],
+#      token         => 'sometoken'
+#      executor      => 'shell',
+#  }
+#
+#  gitlab_ci_multi_runner::runner { "This is My Second Runner":
+#      gitlab_ci_url => 'http://ci.gitlab.examplecorp.com'
+#      tags          => ['tag', 'tag2','npm', 'grunt'],
+#      token         => 'sometoken'
+#      executor      => 'ssh',
+#      ssh_host      => 'cirunners.examplecorp.com'
+#      ssh_port      => 22
+#      ssh_user      => 'mister-ci'
+#      ssh_password  => 'password123'
+#  }
+#
 define gitlab_ci_multi_runner::runner (
     ########################################################
     # Runner Options                                       #
     # Used By all Executors.                               #
     ########################################################
 
-    #URL of the Gitlab Server.
-    $gitlab_ci_url      = undef,
-    #Array of tags
-    $tags               = undef,
-    #CI Token
-    $token              = undef,
-    #Executor - Shell, parallels, ssh, docker etc
-    $executor           = undef,
+    $gitlab_ci_url = undef,
```

Comment:
Puppet style guide recomendation is to align all the hash rockers (=>) , but I didn't see anything explicitely stated about variable declarations alignement and I often find this more annoying than usefull... I want to quickly see what is the value of a specific variable avoiding uneeded empty space. But it is more my personnal taste :) I can align them back if you want (I was pretty sure this one will be problematic ;) ).